### ISSUE TITLE
7$: Disallow parameter named 'await' in async arrow function

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#215]
This code is illegal, but quick-lint-js reports no error for it:
```javascript
(async (await) => null);
```

**Comment by keyehzy** [comment]
I will try this one.

**Comment by strager** [comment]
@keyehzy Any progress?

**Comment by keyehzy** [comment]
I had little progress because of other activities, you may unassign me for now. If I make any significant progress in the next few days and this issue is still open I will contribute.

**Comment by erlliam** [comment]
I claim this for-hire task. I expect payment after I complete this task.

**Comment by singalhimanshu** [comment]
Will this issue fix #241 ?

**Comment by strager** [comment]
> Will this issue fix #241 ?

No. This task (#215) is a subtask of #241, but doesn't cover generator functions or non-arrow functions.
